### PR TITLE
[Comment Editor] New Test: Comment on Notes with addComment Script

### PR DESCRIPTION
**Pull request by noi5e**
Not to be confused with my other current PR, #8801, which adds a new test for Question pages. Please review that one first, it has more extensive write-up!

This test was fairly straightforward to write compared to questions.

I renamed some of the tests to distinguish between notes, questions, and the old tests for commenting on wiki pages.

I simulated a new note in the fixtures for testing comments:

**nodes.yml**
```
comment_note: # for testing comments on notes
  nid: 38
  uid: 2
  title: "Please don't comment here"
  path: "/notes/jeff/12-08-2020/please-don-t-comment-here"
  created: <%= DateTime.new(2020,12,8).to_i %>
  changed: <%= DateTime.new(2020,12,8).to_i %>
  status: 1
  type: "note"
  cached_likes: 0
  slug: "note-please-don-t-comment-here"
```

**revisions.yml**
```
comment_note: # for testing comments on notes
  nid: 38
  uid: 2
  title: "Please don't comment here"
  body: "I beg you, please"
  timestamp: <%= DateTime.new(2020,12,8).to_i %>
  status: 1
```

The test:
```
test 'note: comment via JavaScript, with comment body + URL' do
  visit nodes(:comment_note).path
  page.evaluate_script("addComment('hahaha', '/comment/create/38')")
  assert_selector('#comments-list .comment-body p', text: 'hahaha')
end
```

(This PR is part of the larger Comment Editor Overhaul Project with Outreachy. Refer to Planning Issue #8775 for more context)

### CODE REVIEW

**Review by jywarren — APPROVED** [review]
This looks great, all excellent clarifications and additions. If it runs smoothly in GitPod, let's mark it ready to merge? Hopefully we figure out the testing situation very soon and can merge this!

**Review by pydevsg — APPROVED** [review]
Awesome work @noi5e 🎉

**Comment by noi5e** [comment]
I caught a test that was failing because I created a new fixture:

https://github.com/publiclab/plots2/blob/ff3417ab1ad965753a3a159b905a59b2f244b8ea/test/unit/node_test.rb#L345-L354

Pushed new commits and it fixes the fail!

There might be a few more changes that need to be made. Going through the tests locally right now.

**Comment by noi5e** [comment]
Caught two more tests that were failing bc of the new note fixture:
https://github.com/publiclab/plots2/blob/ff3417ab1ad965753a3a159b905a59b2f244b8ea/test/functional/notes_controller_test.rb#L362-L373
https://github.com/publiclab/plots2/blob/ff3417ab1ad965753a3a159b905a59b2f244b8ea/test/functional/notes_controller_test.rb#L391-L402

27 should be 28 in both because of the new note I created. Pushed the changes! I think this PR is good, no other new failed tests AFAIK locally.

File: test/functional/notes_controller_test.rb [hunk 1]
```diff
@@ -368,7 +368,7 @@ def teardown
 
     assert_response :success
     selector = css_select 'div.note'
-    assert_equal 27, selector.size
+    assert_equal 28, selector.size
```
Comment (Sagarpreet) [inline test/functional/notes_controller_test.rb:371]: Correct 👍

**Review by Sagarpreet — APPROVED** [review]
🎉 🎉 🎉 
All tests are passing now, right?

**Comment by jywarren** [comment]
Awesome, now that #8801 is merged, can you resolve the conflicts noted by GitHub and then we should be good to merge this (bypassing Travis) too? Thank you @noi5e and also @Sagarpreet @pydevsg !!!

**Comment by jywarren** [comment]
Also noting, this is a good example of this happening - perhaps we shouldn't do count-based assertions because it causes interdependence among fixtures. While in some ways it's nice to see the code interdependencies, because it stops us from thinking too narrowly, it also creates barriers to code changes because changes cascade and affect other tested areas. Food for thought! Thanks again!

**Comment by noi5e** [comment]
Just resolved the conflicts!

**Comment by jywarren** [comment]
Hooray!!

**Pull request merged**